libbeat/common/mapstr.go

```diff
@@ -138,6 +138,15 @@ func (m MapStr) Put(key string, value interface{}) (interface{}, error) {
 	return walkMap(key, m, mapStrOperation{putOperation{value}, true})
 }
 
+// SetValue dereferences a value if it is a pointer. Sets the dereferenced value if not nil.
```

Comment:
@andrewkroh I like the suggestion a lot. TBH I was not really aware / forgot that we have this. This removes the hassle of having to think about which method to use when creating the event.

What happens in the case that `Put("a.b.c", nil)` is used and the event is then normalised. Will the full tree be removed?

@simitt In case you agree with the change, do you want to work on this?